jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/statistical/MaxFilter.java

```diff
@@ -21,21 +16,9 @@ public MaxFilter(int maskSize) {
         this.maskSize = maskSize;
     }
 
-    private int maskSize;
-
     @Override
     public void apply(int x, int y) {
-        List<Pixel> pixelList = new ArrayList<>();
-
-        for (int i = -maskSize / 2; i <= maskSize / 2; ++i) {
-            for (int j = -maskSize / 2; j <= maskSize / 2; ++j) {
-                try {
-                    pixelList.add(getPixel(x + i, y + j));
-                } catch (Exception e) {
-                    //corner case, no need to handle
-                }
-            }
-        }
+        List<Pixel> pixelList = getPixelList(x, y);
 
         Integer red = pixelList.stream().map(Pixel::getRed).sorted().max(Comparator.naturalOrder()).get();
```

Comment:
we could create abstract method in StatisticalFilter that would take List<Pixel> and return Pixel. This way all we have to override in max/mean/min/median filters would be this new method instead of whole apply logic. for reference, apply in statistical filter could look like this
apply(int x ,int y){
Pixel p = getPixelResult(getPixelList(x,y));
setPixel(x, y, p);
}